src/Polly.Extensions/Telemetry/TelemetryOptions.cs

```diff
@@ -48,4 +71,12 @@ public class TelemetryOptions
         HttpResponseMessage response => (int)response.StatusCode,
         _ => result,
     };
+
+    /// <summary>
+    /// Gets or sets the resilience event severity provider that allows customizing the severity of resilience events.
+    /// </summary>
+    /// <value>
+    /// The default value is <see langword="null"/>.
+    /// </value>
+    public Func<ResilienceEvent, ResilienceEventSeverity>? SeverityProvider { get; set; }
```

Comment:
```suggestion
    public Func<SeverityProviderArguments, ResilienceEventSeverity>? SeverityProvider { get; set; }
```

Shall we wrap it to allow adding more stuff to it in the future if needed?

cc @martincostello

Reply:
I'm erring to us not needing that as I wonder what we'd do that wasn't inside the event that was in `TelemetryEventArguments<TResult, TArgs>` but that didn't need the type arguments, but at the same time is safer than not if we did just in case.

Reply:
Maybe we can pass `ResilienceContext` for extra flexibility. I wonder whether we are not overdoing it though.